crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -3635,10 +3636,28 @@ impl<'w, 's> FilteredEntityRef<'w, 's> {
             .flatten()
     }
 
+    /// Tries to get access to the component of type `T` for the current entity.
+    ///
+    /// Returns an error distinguishing between insufficient access and the component
+    /// not being present on the entity.
+    #[inline]
+    pub fn try_get<T: Component>(&self) -> Result<&'w T, ComponentFetchError> {
+        let components = self.entity.world().components();
+        let id = components.get_valid_id(TypeId::of::<T>()).ok_or(
+            ComponentFetchError::ComponentNotFound(ComponentId::new(usize::MAX)),
```

Comment:
We might want to differentiate between ComponentNotFound on the entity and ComponentNotRegistered